Add module READMEs and update main README (#11)
WayScience/JUMP-single-cell: README.md

# JUMP-single-cell

## Data

> In this repository, we apply the [phenotypic profiling model](https://github.com/WayScience/phenotypic_profiling_model), which predicts phenotypic class of single cells using nuclei features, on the [JUMP-Target pilot data](https://github.com/jump-cellpainting/JUMP-Target) from the [JUMP consortium](https://jump-cellpainting.broadinstitute.org/).

Reviewer: ```suggestion
In this repository, we apply the [phenotypic profiling model](https://github.com/WayScience/phenotypic_profiling_model), which predicts phenotypic class of single cells using nuclei features, to the [JUMP-Target pilot data](https://github.com/jump-cellpainting/JUMP-Target) from the [JUMP consortium](https://jump-cellpainting.broadinstitute.org/).
```

Author: 👍 Changed

In this dataset, there are 51 plates that are associated with one of three treatment types:

- Compound
- CRISPR
- ORF

Each treatment has it's own platemap and metadata file that can be found in the [reference_plate_data](./reference_plate_data/) folder.
A barcode platemap is include which associates each plate to the correct platemap file.

> There are a total of **20,959,860 single cells** between all plates.

Reviewer: ```suggestion
We segment a total of **20,959,860 single cells** in all plates.
```

Different segmentation algorithms and/or parameters may disagree with this number :)

Author: I might be confused by this one because when I think "we segment" that means we perform the CellProfiler pipeline but that is probably just semantics. I have implemented this change.

> To reproduce this project, please ensure adequate storage as the CellProfiler SQLite database files is approximately 1.1 TB large.

Reviewer: ```suggestion
To reproduce this project, please ensure adequate storage as the CellProfiler SQLite database files are approximately 1.1 TB.
```

Author: 👍 Changed

## Goal

> Traditionally, image-based analyses aggregate single-cell features which removes outliers that could be interesting biology.

Reviewer: ```suggestion
Traditional image-based profiling pipelines aggregate single-cells into well-level profiles.
While, this process removes outliers that might dampen signal, it also removes potentially interesting biologically-meaningful heterogeneity.
```

Author: Thanks for the better clarity! I have added this change!

> By predicting phenotypes for single-cells with the phenotypic profiling model, we hope to uncover patterns of biology that wouldn't be uncovered with the traditional methodology.

Reviewer: ```suggestion
By predicting single-cell phenotypes with our phenotypic profiling model, we hope to uncover important patterns of biology that would be missed with the traditional methodology.

Specifically, the benefits of single-cell phenotyping include:

- Granular phenotypic mechanisms of perturbations regarding (A) the impact perturbations have on a specific phenotype (e.g., disrupting mitosis) and (B) impact on phenotype prevalence (e.g., a gene knockout that causes apoptosis or stalls cells in a specific cell cycle phase)
-  Filter and/or combine cells of the same phenotypic class to purify and/or improve the traditional image-based profiling pipeline
- Adding knowledge to specific combinations of morphology features allows for self-referential interpretation, without the need for database signature lookup or other guilt-by-association methods
- When combined with different experimental designs (e.g., targeted fluorescence marker), we can test specific hypotheses regarding single-cell phenotype distributions (and other important hypotheses that would otherwise be impossible without single-cell phenotypes).
```

Author: I appreciate this in-depth clarification of the importance of single-cell phenotyping!

I have included this!

## Repository Structure

| Module | Purpose | Description |
| :---- | :----- | :---------- |
| [0.download_data](./0.download_data/) | Download JUMP-Target SQLite files | Download CellProfiler SQLite outputs for 51 plates from AWS |
| [1.process_data](./1.process_data/) | Process SQLite files | Using pycytominer, the SQLite outputs are converted to CSV features are merged into single-cells and normalized |
| [2.evaluate_data](./2.evaluate_data/) | Apply phenotypic profiling model | Phenotypic predictions are generated for single-cells using the phenotypic profiling model |
| [3.analyze_data](./3.analyze_data/) | Analyze phenotypic predictions | Multiple analyses are performed to validate the phenotypic predicted class for each treatment compared to control |

> | [reference_plate_data](./reference_plate_data/) | Platemaps per treatment type | This folder holds the platemap files with metadata based on treatment and the barcode platemap file |

Reviewer: There is some passive voice in this table, please update to active voice :)

Author: I have corrected this! Thanks for pointing it out!

> ## Main environment

Reviewer: ```suggestion
## Computational environment
```

Is this what you mean? I wasn't sure

Author: I have used this in prior repos to designate there is one main environment to use for all modules. I can add computational if it makes it clearer.
